src/main/java/bio/terra/workspace/app/controller/WorkspaceApiController.java

```diff
@@ -63,16 +66,27 @@ private AuthenticatedUserRequest getAuthenticatedInfo() {
     WorkspaceStageModel requestStage = body.getStage();
     requestStage = (requestStage == null ? requestStage.RAWLS_WORKSPACE : requestStage);
     WorkspaceStage internalStage = WorkspaceStage.fromApiModel(requestStage);
-    return new ResponseEntity<>(
-        workspaceService.createWorkspace(body, internalStage, userReq), HttpStatus.OK);
+    Optional<SpendProfileId> spendProfileId =
+        Optional.ofNullable(body.getSpendProfile()).map(SpendProfileId::create);
+
+    UUID createdId =
+        workspaceService.createWorkspace(body.getId(), spendProfileId, internalStage, userReq);
```

Comment:
consider whether `createWorkspace` should take `(Workspace workspace, AuthenticatedUserRequest userReq)` as it's arguments. similarly for workspaceDao.createWorkspace. WorkspaceService API already includes the 'Workspace' type; right now there's nothing else needed or extraneous to creating a workspace. That could change later (more likely in WorkspaceService than in WorkspaceDao), but the parallel atm is correct.

Reply:
I was mainly thinking about this from the perspective of "will the controller always have all information needed to make a workspace?" If not (e.g. if we generate something midway through a flight), I'm not thrilled with passing around a slightly-invalid workspace object. That doesn't really hold for workspaceDao though, I'll change it there.

Reply:
Totally agree on the slightly invalid workspace object being bad idea. It may be worth creating a WorkspaceRequest kind of internal rep that's distinct from the `Workspace` class when we get there, but I can't bring myself to have it yet when they don't differ. But yes, does need discipline later and agreed on workspaceDao.